docs/source/getting-started.md

```diff
@@ -0,0 +1,201 @@
+
+- Python 3.10 - 3.11
+- g++ version 7 or higher, clang++ version 5 or higher, or another C++17-compatible toolchain.
+- Linux or MacOS operating system (Arm or x86).
+  - Windows is supported via WSL.
+
+<hr/>
+
+## Preparing the Model
+Exporting is the process of taking a PyTorch model and converting it to the .pte file format used by the ExecuTorch runtime. This is done using Python APIs. PTE files for common models can be found on HuggingFace (TODO add link).
+
+### Requirements
+- A PyTorch model.
+- Example model inputs, typically as PyTorch tensors. You should be able to successfully run the PyTorch model with these inputs.
+- One or more target hardware backends.
+
+### Selecting a Backend
+ExecuTorch provides hardware backends for a wide variety of hardware. The most commonly used backends are XNNPACK, for Arm and x86 CPU, CoreML (for iOS), Vulkan (for Android GPUs), and Qualcomm (for Qualcomm-powered Android phones).
+
+For mobile use cases, consider using XNNPACK for Android and CoreML or XNNPACK for iOS as a first step. See [Delegates](/TODO.md) for a description of available backends.
+
+### Exporting
+Exporting is done using Python APIs. ExecuTorch provides a high degree of customization during the export process, but the typical flow is as follows:
+```python
+model = MyModel() # The PyTorch model to export
+example_inputs = (torch.randn(1,3,64,64),) # A tuple of inputs
+
+et_program =
+ executorch.exir.to_edge_transform_and_lower(
+ torch.export.export(model, example_inputs)
+partitioner=[XnnpackPartitioner()]
+).to_executorch()
+
+with open(“model.pte”, “wb”) as f:
+	f.write(et_program.buffer)
+```
+
+If the model requires varying input sizes, you will need to specify the varying dimensions and bounds as part of the `export` call. See [Exporting a Model for ExecuTorch](/TODO.md) for more information.
+
+The hardware backend to target is controlled by the partitioner parameter to to\_edge\_transform\_and\_lower. In this example, the XnnpackPartitioner is used to target mobile CPUs. See the delegate-specific documentation for a full description of the partitioner and available options.
+
+Quantization can also be done at this stage to reduce model size and runtime. Quantization is backend-specific. See the documentation for the target backend for a full description of supported quantization schemes.
+
+### Testing the Model
+
+After successfully generating a .pte file, it is common to use the Python runtime APIs to validate the model on the development platform. This can be used to evaluate model accuracy before running on-device. 
+
+Inference can be run as follows:
+```python from executorch.extensions.pybindings.portable_lib import _load_for_executorch
+
+input_tensor = torch.randn(1,3,128,128)
+model = _load_for_executorch(“/path/to/model.pte”)
+outputs = model(input_tensor)
```

Comment:
these are legacy pybinding libraries. Use the new stuff

https://colab.research.google.com/drive/1qpxrXC3YdJQzly3mRg-4ayYiOjC6rue3?usp=sharing